aten/src/ATen/core/TensorMethods.h

```diff
@@ -57,1240 +58,1652 @@ inline void Tensor::set_data(Tensor new_data) {
 
 // all static inline to allow for inlining of the non-dynamic part of dispatch
 inline Tensor Tensor::abs() const {
-    return dispatch_type().abs(*this);
+    return globalATenDispatch().getOp<Tensor (const Tensor &)>(
+        tensorTypeIdToBackend(type_id()), is_variable(), 30, "abs")(*this);
```

Comment:
The hard coded numbers here make me concerned about merge conflicts that occur when these numbers wobble. If I insert a new function, that will shift the numbers for everything allocated after it down one; if two people edit unrelated functions, this will assuredly cause merge conflicts from renumbering. (Also, are we **sure** that the number generation process here is deterministic? It'll be really bad if someone else runs the generator script and the numbering is different.)

Reply:
A very most basic mitigation I can think of is to ensure the numbering only occurs in one place, so at least you only have one file to solve merge conflicts on. An even better mitigation is to ensure this file is NOT checked in (to do that, you'll have to solve the "we don't know how to codegen in xplat" problem one way or another.)

Reply:
Yes, we can move it out (I think usages in caffe2/core/operator.h are already ifdefed or can be ifdefed). We'd still need to figure out xplat codegen at some point for PT mobile track. But it's not an immediate blocker.

Reply:
Let's go with the solution we discussed offline - a pointer to the dispatch table, looked up by string and stored in a static variable in the function so it's only looked up once. That seems to be the easiest solution.